ghga_service_chassis_lib/s3.py

```diff
@@ -172,6 +172,46 @@ def _translate_s3_client_errors(
     return exception
 
 
+def adapt_part_size(current_part_size: int, file_size: int) -> int:
+    """
+    Checks if the provided part size is within bounds and produces <= 10_000 parts.
+
+
+    Args:
+        current_part_size (int): currently chosen part size in bytes to be checked
+        file_size (int): total file size in bytes
+
+    Raises:
+        ValueError if file size exceeds the maximum of 5 TiB
+    """
+    lower_bound = 5 * 1024**2
+    upper_bound = 5 * 1024**3
+    hard_upper_limit = 5 * 1024**4
+    max_num_parts = 10_000
```

Comment:
PMO: You may consider to make them global constants and move them out of the function.